### PR TITLE
Diagnose and prevent Svelte effect depth loops

### PR DESCRIPTION
**Pull request by ejsmith**
## Summary

- add bounded production diagnostics for `effect_update_depth_exceeded`
- append repeated reactive state-write call sites to the captured error stack so existing source maps resolve application locations
- attach structured `svelte_effect_depth` extended data with update counts, effect ancestry, pathname, visibility, and safe interaction metadata
- update `svelte-time` from 2.3.0 to 2.3.1 for its adaptive live-interval loop fix
- retain adaptive relative-time updates and add a deterministic regression test covering shared clocks on opposite sides of the one-hour boundary

## Root cause

`svelte-time@2.3.0` computed its adaptive interval from a shared cached `now` value selected by that same interval. When active shared clocks straddled an age boundary, a live timestamp could alternate between ticker tiers until Svelte threw `effect_update_depth_exceeded`.

The defect was reported with a standalone reproduction in [metonym/svelte-time#107](https://github.com/metonym/svelte-time/issues/107) and fixed upstream in [svelte-time 2.3.1](https://github.com/metonym/svelte-time/releases/tag/v2.3.1). We removed the application-level fixed-minute workaround and restored `live={true}` after upgrading.

## Diagnostics

The production instrumentation mirrors the useful part of Svelte's development-only state-write tracking while keeping collection bounded:

- starts stack collection only after repeated writes to the same state source
- limits sources, unique stacks, captures, and appended frames
- does not capture state values, query strings, DOM text, or user input
- fails the build if the supported Svelte runtime version or expected internal source shape changes

A minified production preview with an intentional reactive loop captured 1,001 writes and resolved the generated frame through the production source map to the exact Svelte assignment.

## Dependency review

- reviewed the complete 2.3.0...2.3.1 source diff; the runtime change uses a fresh `dayjs()` value for adaptive tier selection in `Time` and `Duration`
- no API or migration changes
- same-day patch release risk is covered by the upstream boundary regressions and our application-level deterministic regression
- `npm audit` still reports the same 6 existing findings (1 low, 1 moderate, 4 high); this update introduced no additional findings

## Validation

- deterministic formatter regression fails with 2.3.0 and passes with 2.3.1 while keeping adaptive `live={true}` behavior
- focused diagnostics tests: 4 passed
- `npm run validate`
- `npm run test:unit -- --run`
- `npm run build`

## Breaking changes

None.

### CODE REVIEW

File: src/Exceptionless.Web/ClientApp/src/lib/telemetry/svelte-effect-depth-diagnostics.ts [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+function recordStateUpdate(source: unknown) {
+    if (!stateSourceUpdates || (typeof source !== 'object' && typeof source !== 'function') || source === null) {
+        return;
+    }
+
+    totalStateUpdates++;
+    let updates = stateSourceUpdates.get(source);
+    if (!updates) {
+        updates = { captureCount: 0, count: 0, stacks: new Map() };
+        stateSourceUpdates.set(source, updates);
```
[the remark this answers is not in the document]
Comment (ejsmith) [reply src/Exceptionless.Web/ClientApp/src/lib/telemetry/svelte-effect-depth-diagnostics.ts:192]: Fixed in 4a02cfdf3. `recordStateUpdate` now stops allocating new source entries once `MAX_STATE_SOURCES` is reached while still incrementing `totalStateUpdates`. Added a high-fan-out regression that records 200 writes across 101 sources and verifies only 8 source entries are retained; the focused test and full validate/unit/build suite pass.
